From ac5b6c6906550746e6b0b02066ffbaa6fdbf3b4b Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Tue, 31 Mar 2015 18:17:35 +0800
Subject: [PATCH 0292/5242] MLK-10515-2: ASoC: fsl_mqs: Move clk get_rate to
 hw_param

commit  e1a4b1691f4cb2e5adb6b818753ce11b594903e6 from
https://source.codeaurora.org/external/imx/linux-imx.git

It is too early to put clk get rate in probe, because the rate for
the clock may not be ready.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/codecs/fsl_mqs.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/sound/soc/codecs/fsl_mqs.c b/sound/soc/codecs/fsl_mqs.c
index 597306e..2102391 100644
--- a/sound/soc/codecs/fsl_mqs.c
+++ b/sound/soc/codecs/fsl_mqs.c
@@ -1,7 +1,7 @@
 /*
  * ALSA SoC IMX MQS driver
  *
- * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ * Copyright (C) 2014-2015 Freescale Semiconductor, Inc.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
@@ -46,6 +46,8 @@ static int fsl_mqs_hw_params(struct snd_pcm_substream *substream,
 
 	int div, res;
 
+	mqs_priv->mclk_rate = clk_get_rate(mqs_priv->mclk);
+
 	mqs_priv->bclk = snd_soc_params_to_bclk(params);
 	mqs_priv->lrclk = params_rate(params);
 
@@ -186,8 +188,6 @@ static int fsl_mqs_probe(struct platform_device *pdev)
 		goto out;
 	}
 
-	mqs_priv->mclk_rate = clk_get_rate(mqs_priv->mclk);
-
 	dev_set_drvdata(&pdev->dev, mqs_priv);
 
 	return devm_snd_soc_register_component(&pdev->dev, &soc_component_fsl_mqs,
-- 
1.7.9.5

